language/move-core/types/src/vm_status.rs

```diff
@@ -235,9 +235,26 @@ pub fn convert_prologue_runtime_error(status: VMStatus) -> VMStatus {
             };
             VMStatus::Error(new_major_status)
         }
-        status @ VMStatus::ExecutionFailure { .. }
-        | status @ VMStatus::Error(_)
+        status @ VMStatus::ExecutionFailure { .. } | status @ VMStatus::Executed => status,
+        VMStatus::Error(code) => {
+            debug_assert!(false, "Unexpected prologue/epilogue error: {:?}", code);
```

Comment:
we should log critical here. We can do that at another time if needed. Just wanted to get consensus on that

Reply:
Logging was a lot easier once I moved the errors. So its up on the latest revision :)